apis/condition_set.go

```diff
@@ -242,23 +242,38 @@ func (r conditionsImpl) ClearCondition(t ConditionType) error {
 // MarkTrue sets the status of t to true, and then marks the happy condition to
 // true if all other dependents are also true.
 func (r conditionsImpl) MarkTrue(t ConditionType) {
-	// set the specified condition
+	// Save the original status of t.
+	org := r.GetCondition(t).DeepCopy()
+	orgTL := r.GetTopLevelCondition().DeepCopy()
+
+	// Set the specified condition.
 	r.SetCondition(Condition{
 		Type:     t,
 		Status:   corev1.ConditionTrue,
 		Severity: r.severity(t),
 	})
 
-	// check the dependents.
+	// Check the dependents.
 	for _, cond := range r.dependents {
 		c := r.GetCondition(cond)
-		// Failed or Unknown conditions trump true conditions
+		// Failed or Unknown conditions trump true conditions.
 		if !c.IsTrue() {
+			// Update the happy condition if the current ready condition is
```

Comment:
Why do this comparison, rather than always updating this?

Reply:
because you want the last status to change bad to be reflected, not the first you find.